crates/virtio-queue/src/lib.rs

```diff
@@ -12,9 +12,10 @@
 
 //! A crate that exposes the virtio queue API.
 
-#![deny(missing_docs)]
+// #![deny(missing_docs)]
 
 pub mod defs;
+pub mod mock;
```

Comment:
This sub module should be conditionally enabled for test only.

Reply:
I wanted to add just a few more changes to `MockSplitQueue`. After that I'll squash some of the commits, add some documentation and I'll leave the module behind a dev-dependency enabled feature, as suggested.

Reply:
Yeah it looks like some of the commits can (and should :D) be squashed. Also, I think we don't need to leave the module as a dev-dependency enabled feature really; we can have a feature that enables the module for external consumers, and then conditionally compile it with something `like #[cfg(any(test, feature="feature_name"))]` (haven't actually validated that's the correct syntax). That way, it's enabled for external consumers when the feature is selected, and when tests run within the local crate.